crates/core/Cargo.toml

```diff
@@ -66,7 +66,7 @@ tokio = { workspace = true, features = [
     "fs",
     "parking_lot",
 ] }
-
+async-channel = { workspace = true }
```

Comment:
Since this is only required for the datafusion write, I am comfortable adding the new dependency but would like it to be optional and gated/included with the `datafusion` feature